core/domain/question_services.py

```diff
@@ -217,3 +224,254 @@ def update_question(
+    """Gets skill summaries for the goven skills.
+
+    Args:
+        skill_ids: list(str). The list of all linked skill ids.
+
+    Returns:
+        skill_summaries: list(SkillSummary). The skill summary for a skill.
+    """
+    skill_summaries = []
+    linked_skill_summaries = (
+        skill_models.SkillSummaryModel.get_multi(skill_ids, strict=False))
+
+    for linked_skill_summary in linked_skill_summaries:
+        skill_summaries = (
+            skill_services.get_skill_summary_from_model(linked_skill_summary))
+
+    return skill_summaries
+
+
+def get_question_rights_from_model(question_rights_model):
+    """Constructs a QuestionRights object from the given question rights model.
+
+    Args:
+        question_rights_model: QuestionRightsModel. Question rights from the
+            datastore.
+
+    Returns:
+        QuestionRights. The rights object created from the model.
+    """
+
+    return question_domain.QuestionRights(
+        question_rights_model.id,
+        question_rights_model.creator_id
+    )
+
+
+def create_new_question_rights(question_id, committer_id):
+    """Creates a new question rights object and saves it to the datastore.
+
+    Args:
+        question_id: str. ID of the question.
+        committer_id: str. ID of the committer.
+    """
+    question_rights = question_domain.QuestionRights(
+        question_id, committer_id)
+    commit_cmds = [{'cmd': question_domain.CMD_CREATE_NEW}]
+
+    question_models.QuestionRightsModel(
+        id=question_rights.id,
+        creator_id=question_rights.creator_id
+    ).commit(committer_id, 'Created new question rights', commit_cmds)
+
+
+def get_question_rights(question_id, strict=True):
+    """Retrieves the rights object for the given question.
+
+    Args:
+        question_id: str. ID of the question.
+        strict: bool. Whether to fail noisily if no question rights with a
+            given id exists in the datastore.
+
+    Returns:
+        QuestionRights. The rights object associated with the given question.
+
+    Raises:
+        EntityNotFoundError.The question rights for question with ID
+            question_id was not found in the datastore.
+    """
+
+    model = question_models.QuestionRightsModel.get(
+        question_id, strict=strict)
+
+    if model is None:
+        return None
+
+    return get_question_rights_from_model(model)
+
+
+def get_all_question_rights():
+    """Returns the rights object of all questions present in the datastore.
+
+    Returns:
+        dict. The dict of rights objects of all questions present in
+            the datastore keyed by question id.
+    """
+    question_rights_models = question_models.QuestionRightsModel.get_all()
+    question_rights = {}
+    for model in question_rights_models:
+        rights = get_question_rights_from_model(model)
+        question_rights[rights.id] = rights
+    return question_rights
+
+
+def check_can_edit_question(user, question_rights):
```

Comment:
Why is this here and not in a file related to ACLs? I don't see a similar function in exp_services, which makes me think that this might be in the wrong place.